base_layer/core/src/consensus/consensus_manager.rs

```diff
@@ -166,6 +181,9 @@ impl ConsensusManagerBuilder {
             network: self.network,
             emission,
             gen_block: self.gen_block,
+            chain_strength_comparer: self
+                .chain_strength_comparer
+                .unwrap_or(strongest_chain().by_accumulated_difficulty().build()),
```

Comment:
`strongest_chain()....` should only run if required.
```suggestion
                .unwrap_or_else(|| strongest_chain().by_accumulated_difficulty().build()),
```